st2api/st2api/controllers/v1/webhooks.py

```diff
@@ -110,6 +115,21 @@ def post(self, *args, **kwargs):
 
         return body
 
+    def _parse_request_body(self, content_type, body):
+        if content_type == 'application/json':
+            self._log_request('Parsing body as JSON', request=pecan.request)
+            body = json.loads(body)
+        elif content_type in ['application/x-www-form-urlencoded', 'multipart/form-data']:
+            self._log_request('Parsing body as form encoded data', request=pecan.request)
+            body = urlparse.parse_qs(body)
```

Comment:
Heh, I was actually just thinking about the same thing so I made the change above to capture all the exception types.